### PR TITLE
Added factorization and seeders for data

### PR DESCRIPTION
**Pull request by txmxrxrxnxi**

### CODE REVIEW

File: src/DataImport/Models/ReadingProgress.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	public int ID { get; set; }
+	public int BookID { get; set; }
+	public int UserID { get; set; }
+	public int LastPage { get; set; }
+	public bool IsFavourite { get; set; }
+	public BookTheme Theme { get; set; }
+	public int FontSize { get; set; }
+	public DateTime UpdatedAt { get; set; }
+
+	public ReadingProgress()
```
Comment (lolpie244) [inline src/DataImport/Models/ReadingProgress.cs:24]: you can define default values for property, so there is no need in constructor

File: src/DataImport/Models/User.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	public DateTime UpdatedAt { get; set; }
+
+	public User()
+		: this(id: 0, email: "", username: "", password: "")
+	{
+		return;
+	}
+	
+	public User(
+		int id,
```
Comment (lolpie244) [inline src/DataImport/Models/User.cs:31]: we don't need a constructor if we don't have additional logic. Better to use `new User(){field = value}`

File: src/DataImport/Utilities/Hashing.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+		{
+			throw new ArgumentException(
+				message: "Password cannot be null");
+		}
+		
+		byte[]? hash = SHA256.Create().ComputeHash(
+				buffer: Encoding.UTF8.GetBytes(s: pass));
+		StringBuilder str = new StringBuilder();
+
+		foreach (byte b in hash)
```
Comment (lolpie244) [inline src/DataImport/Utilities/Hashing.cs:21]: use `BitConverter.ToString(hash)`, instead foreach

File: src/DataImport/Utilities/Seeders.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+using DataImport.Models;
+using Faker;
+
+namespace DataImport.Utilities;
+
+public static class Seeders
+{
+	public static int length = 38;
+	
+	public static Book[] GetBooks()
```
Comment (lolpie244) [inline src/DataImport/Utilities/Seeders.cs:11]: Why do you separate books from other seeds?

**Review by lolpie244 — CHANGES_REQUESTED** [review]

File: src/DataImport/Utilities/Seeders.cs (hunk 4 above)
Comment (txmxrxrxnxi) [reply src/DataImport/Utilities/Seeders.cs:11]: Those are not random books, those are set ones with info from files

File: src/DataImport/Utilities/Hashing.cs (hunk 3 above)
Comment (txmxrxrxnxi) [reply src/DataImport/Utilities/Hashing.cs]: +

File: src/DataImport/Models/User.cs (hunk 2 above)
Comment (txmxrxrxnxi) [reply src/DataImport/Models/User.cs]: `+

File: src/DataImport/Utilities/Seeders.cs (hunk 4 above)
Comment (lolpie244) [reply src/DataImport/Utilities/Seeders.cs:11]: so what? Our other random data is also seeds. So it's better to not separate books and other seeds. You can move other "random" seeds into this file

File: src/DataImport/Models/ReadingProgress.cs (hunk 1 above)
Comment (txmxrxrxnxi) [reply src/DataImport/Models/ReadingProgress.cs]: `+

File: src/DataImport/Utilities/Seeders.cs (hunk 4 above)
Comment (txmxrxrxnxi) [reply src/DataImport/Utilities/Seeders.cs:11]: Random data are not the same the seeders are

Comment (lolpie244) [reply src/DataImport/Utilities/Seeders.cs:11]: Oh, yeah, I forgot about it

**Review by lolpie244 — APPROVED** [review]

File: src/DataImport/Factories/BookFactory.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+	: ModelFactory<Book>
+{
+	public int Id { get; } = RandomNumber.Next();
+	public int PublisherId { get; } = RandomNumber.Next();
+	public string Author { get; } = Faker.Name.FullName();
+	public string Name { get; } = Company.Name();
+	public string Genre { get; } = Lorem.GetFirstWord();
+	public string Description { get; } = Lorem.Paragraph();
+	public float Rating { get; } = RandomNumber.Next() % 6;
+	public int PagesNumber { get; } = RandomNumber.Next();
```
Comment (lolpie244) [inline src/DataImport/Factories/BookFactory.cs:16]: "UrlToFIle" field missing

**Review by lolpie244 — CHANGES_REQUESTED** [review]

File: src/DataImport/Factories/BookFactory.cs (hunk 5 above)
Comment (txmxrxrxnxi) [reply src/DataImport/Factories/BookFactory.cs:16]: `+

File: src/DataImport/Models/Bookmark.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+
+namespace DataImport.Models;
+
+
+public class Bookmark
+{
+	public int Id { get; set; }
+	public int BookId { get; set; }
```
Comment (Combo3zoom) [inline src/DataImport/Models/Bookmark.cs]: change foreign key bookId and UserId to readingProgressId

**Review by lolpie244 — APPROVED** [review]

File: src/DataImport/Models/Bookmark.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+namespace DataImport.Models;
+
+
+public class Bookmark
+{
+	public int Id { get; set; }
+	public int BookId { get; set; }
+	public int UserId { get; set; }
+	public int PageNumber { get; set; }
+	public string Note { get; set; }
```
Comment (Combo3zoom) [inline src/DataImport/Models/Bookmark.cs:11]: delete prop Note we don't use him

File: src/DataImport/Models/Bookmark.cs (hunk 6 above)
Comment (txmxrxrxnxi) [reply src/DataImport/Models/Bookmark.cs]: `+

File: src/DataImport/Models/Bookmark.cs (hunk 7 above)
Comment (txmxrxrxnxi) [reply src/DataImport/Models/Bookmark.cs]: `+

Comment (lolpie244) [reply src/DataImport/Models/Bookmark.cs]: no, we use note

**Pull request merged**